lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4400,9 +4400,8 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     // Disable the buffer's top-level cleanup (which is supposed to operate on
     // the buffer), create a cleanup for the value that carrys all child
     // cleanups.
-    auto valueCleanup = makeCleanup(adjVal, emitCleanup,
-        adjBuf.getCleanup()
-            ? adjBuf.getCleanup()->getChildren() : ArrayRef<Cleanup *>());
```

Comment:
i think this should be reverted. no?

Reply:
Hmm, I think we may need this, since `adjBuf` no longer gets a new cleanup.
```diff
-    auto cleanup = makeCleanup(adjBuf, emitCleanup);	
-    adjBuf.setCleanup(cleanup);
```

Reply:
Because there's no cleanup for accesses,`adjBuf.getCleanup()` is always null.